python/ray/data/tests/test_dataset.py

```diff
@@ -604,6 +604,43 @@ def apply_logical_ops(arr):
     np.testing.assert_equal(apply_logical_ops(arr), apply_logical_ops(df["two"]))
 
 
+def test_tensor_array_array_protocol(ray_start_regular_shared):
```

Comment:
It looks there are no tests for pandas.py which is concerning. Can we start a test file for it under ray/air/... and move these two tests over?

Reply:
`pandas.py` is covered by these tests! When the move to `ray/air/...` happened, they split the Pandas extension type and the Arrow extension type into two files, but they're both still imported through `ray/data/extensions/tensor_extension.py` and they're both covered by these tests.